RxSwift/Observables/ObserveOn.swift

```diff
@@ -101,7 +101,36 @@ final fileprivate class ObserveOnSink<O: ObserverType> : ObserverBase<O.E> {
             _scheduleDisposable.disposable = self._scheduler.scheduleRecursive((), action: self.run)
         }
     }
-    
+
+    #if swift(>=3.2)
+    func run(_ state: Void, recurse: () -> Void) {
```

Comment:
```
/Users/alan/repositories/github/RxSwift/RxSwift/Observables/ObserveOn.swift:105:38: warning: when calling this function in Swift 4 or later, you must pass a '()' tuple; did you mean for the input type to be '()'?
    func run(_ state: Void, recurse: (Void) -> Void) {
                                     ^~~~~~
```

I would have liked to write this as just

```
#if swift(>=3.2)
func run(_ state: Void, recurse: () -> Void) {
#else
func run(_ state: Void, recurse: (Void) -> Void) {
#endif
```

But the compiler doesn't allow that.